aws/flatcar-linux/kubernetes/cl/controller.yaml.tmpl

```diff
@@ -156,6 +157,48 @@ storage:
             --net=host \
             --dns=host \
             --exec=/bootkube -- start --asset-dir=/assets "$@"
+    - path: /opt/etcd-rejoin
+      filesystem: root
+      mode: 0555
+      contents:
+        inline: |
+          #!/bin/bash
+          set -eou pipefail
+          # Script to rejoin as fresh node to an existing etcd cluster when the node cannot join.
+          # The script should be added to /etc/systemd/system/etcd-member.service.d/40-etcd-cluster.conf
+          # as ExecStopPost=-/opt/etcd-rejoin because it should run when the etcd service failed and because it
+          # makes use of the environment variables set there for etcd-member.service.
+          # It only runs when etcd failed to join the cluster (e.g., after repovisioning the node or after
+          # the node was down for too long and got kicked out).
+          # Skip if node was not fully provisioned yet.
+          if [ ! -d "/etc/ssl/etcd/" ]; then exit 0; fi
```

Comment:
But `exit 0` will cause this to be re-executed later?

Reply:
Yes, the systemd service is scheduled for restart.

Reply:
:ok_hand: